lib/storage/acl.js

```diff
@@ -328,3 +493,90 @@ Acl.prototype.makeReq_ = function(method, path, query, body, callback) {
 };
 
 module.exports = Acl;
+
+/**
+ * Attach functionality to a {module:storage/acl} instance. This will add an
+ * object for each role group (owners, readers, and writers), with each object
+ * containing methods to add or delete a type of entity.
+ *
+ * As an example, here are a few methods that are created.
+ *
+ *   myBucket.acl.readers.deleteGroup('groupId', function(err) {});
+ *
+ *   myBucket.acl.owners.addUser('email@example.com', function(err) {});
+ *
+ *   myBucket.acl.writers.addDomain('example.com', function(err) {});
+ *
+ * @private
+ */
+function AclRoleAccessorMethods() {
+  AclRoleAccessorMethods.roles.forEach(this._assignAccessMethods.bind(this));
+}
+
+AclRoleAccessorMethods.accessMethods = [
+  'add',
+  'delete'
+];
+
+AclRoleAccessorMethods.entities = [
+  // Special entity groups that do not require further specification.
+  'allAuthenticatedUsers',
+  'allUsers',
+
+  // Entity groups that require specification, e.g. `user-email@example.com`.
+  'domain-',
+  'group-',
+  'project-',
+  'user-'
+];
+
+AclRoleAccessorMethods.roles = [
+  'owner',
+  'reader',
+  'writer'
+];
+
+AclRoleAccessorMethods.prototype._assignAccessMethods = function(role) {
+  var that = this;
+
+  var accessMethods = AclRoleAccessorMethods.accessMethods;
+  var entities = AclRoleAccessorMethods.entities;
+  var roleGroup = role + 's';
+
+  this[roleGroup] = entities.reduce(function(acc, entity) {
+    var isPrefix = entity.substr(-1) === '-';
+
+    accessMethods.forEach(function(accessMethod) {
+      var method = accessMethod + entity[0].toUpperCase() + entity.substr(1);
+
+      if (isPrefix) {
+        method = method.replace('-', '');
+      }
+
+      // Wrap the parent accessor method (e.g. `add` or `delete`) to avoid the
+      // more complex API of specifying a `scope` and `role`.
+      acc[method] = function(entityId, callback) {
+        var scope;
+
+        if (isPrefix) {
+          scope = entity + entityId;
+        } else {
+          // If the scope is not a prefix, it is a special entity group that
+          // does not require further details. The accessor methods only accept
+          // a callback.
+          scope = entity;
+          callback = entityId;
+        }
+
+        that[accessMethod]({
+          scope: scope,
+          role: role
```

Comment:
Judging from the API working in the regression tests, it's not required to be uppercase. Nonetheless, it's an oversight that I didn't. Fixed: https://github.com/GoogleCloudPlatform/gcloud-node/commit/aefdd45899ee4e6ec2bd012ff3e195bf9ebbdcf3